pkg/asset/installconfig/azure/validation.go

```diff
@@ -161,6 +163,30 @@ func validateInstanceTypes(client API, ic *types.InstallConfig) field.ErrorList
 	return allErrs
 }
 
+// validateDiskEncryption checks if the specified disk encryption is valid.
+func validateDiskEncryption(client API, ic *types.InstallConfig) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if ic.Platform.Azure == nil || ic.Platform.Azure.DefaultMachinePlatform == nil {
+		return allErrs
+	}
+
+	encryptionAtHost := ic.Platform.Azure.DefaultMachinePlatform.EncryptionAtHost
+	if ic.Azure.CloudName == aztypes.StackCloud && encryptionAtHost == true {
+		return append(allErrs, field.Invalid(field.NewPath("encryptionAtHost"), encryptionAtHost, "encryption at host is not supported not this platform"))
+	}
+
+	encryptionType := ic.Platform.Azure.DefaultMachinePlatform.OSDisk.EncryptionType
+	if ic.Azure.CloudName == aztypes.StackCloud && encryptionType != "" {
+		return append(allErrs, field.Invalid(field.NewPath("encryptionType"), encryptionType, "customer managed keys encryption is not supported on this platform"))
+	}
+
+	if encryptionAtHost == true && encryptionType != string(azcompute.EncryptionAtRestWithCustomerKey) {
+		return append(allErrs, field.Invalid(field.NewPath("encryptionType"), encryptionType, "encryption at host is only supported with customer managed keys"))
+	}
```

Comment:
This is not correct. Encryption at host can be enabled independently of server side encrpytion (encryption at rest) and doesn't require customer managed keys ([example in docs](https://docs.microsoft.com/en-us/azure/virtual-machines/linux/disks-enable-host-based-encryption-cli#create-a-vm-with-encryption-at-host-enabled-with-platform-managed-keys)). It is a separate feature. Just a bool on a VM.

However it can be used together with server side encrpytion ([example in docs](https://docs.microsoft.com/en-us/azure/virtual-machines/linux/disks-enable-host-based-encryption-cli#create-a-vm-with-encryption-at-host-enabled-with-customer-managed-keys)). In that case temp disks are still encrypted with platform-managed keys.

Reply:
[This document](https://docs.microsoft.com/en-us/azure/virtual-machines/disk-encryption-overview) will either help you to get better understanding or confuse you more :) My journey with these docs was like this: I have no idea what I'm doing -> Ok, I think I understnad it now -> Nope, I'm lost again -> Ok, I think I finally understand it.

Hopefully this sumarry will help. There are three things related to disk encryption:

* Azure Disk Encryption (out of scope)
* Server-Side Encryption or encryption at rest
  * Enabled by default with platform managed keys
  * Customer can provide their own keys (this is what we want to achive here for our OCP/ARO clusters)
* Encryption at host
  * DIsabled by default
  * Can be enabled separately (in scope)
  * Can be enabled together with Server-Side Encryption with patform managed keys (in scope)
  * Can be enabled together with Server-Side Encryption with customer managed keys (in scope)

When I say in scope/not in scope - I refer to what I originally tried to do in #5134

Reply:
Thanks for the clarification. My understanding was SSE is always on, no customer managed keys required. Encryption at host required customer managed keys (I was not able to find clarification on this in the docs). If encryption at host does not require customer managed keys, I will update things. Thanks for pointing these things out.